packages/outputarea/src/widget.ts

```diff
@@ -250,6 +251,19 @@ class OutputArea extends Widget {
     }
   }
 
+  /**
+   * Follow changes on the model state.
+   */
+  protected onModelStateChanged(sender: IOutputAreaModel, model: IOutputModel): void {
```

Comment:
Perhaps rather than changing the signal type, we can just access `this.model` here, and not worry about more granular changes to the output model.

Reply:
Ya, my original PR simply did the following at https://github.com/jupyterlab/jupyterlab/blob/master/packages/outputarea/src/widget.ts#L404:

```js
model.stateChanged.connect(() => {
  widget.renderModel(model);
})
```

This was simpler but also seemed more like a hack than this approach. If you have a better idea of how to accomplish this (update an output in response to a change in its model vs. just when a cell is re-run), then feel free to make changes to this branch.

Reply:
Another thing is that `this.model` here is `IOutputAreaModel` but we need `IOutputModel` to call `widget.rendeModel` with...